### PR TITLE
Render connection matrix using dash table

### PR DESCRIPTION
**Pull request by mateuszmidor**
KNTK-294

- display measurement data with coloring
- detailed data in hover text
- navigate to timeseries view when clicked on specific cell
- support 60 x 60 matrix

Removed Jinja2

### CODE REVIEW

**Review by mmac-m3a — CHANGES_REQUESTED** [review]
It is mostly good! We though need to either make agent labels configurable or use `agent.name` (see my more detailed comment). Agent aliases are duplicated. For example agents 1022 and 662, both have alias "Frankfurt, Germany". With my fix taking names from the `mesh` section of augmented health results, the first one should have `name` "ibm, eu-de" and the latter "Frankfurt am Main, Choopa,US" (which is confusing, but we cannot fix that).
I've only now noticed that the "agent name fix" does not work in the current version at HEAD in master. For example, agent 1022 has label "eu-de" in the matrix (should be "ibm, eu-de"). I've checked the health data returned for my mesh test by the API, and it is all there.  Looks like your refactor adding `MeshConfig` somehow broke it. I'm not sure how, but it did work before.

File: presentation/matrix_view.py [hunk 1]
```diff
@@ -273,54 +256,10 @@ def make_cell_hover_text(self, from_agent: Agent, to_agent: Agent, mesh: MeshRes
         health = conn.latest_measurement
         if health:
             for m in MetricType:
-                cell_hover_text.append(f"{m.value}: {self.format_health(m, health, True)}")
-            cell_hover_text.append(f"Time stamp: {health.timestamp.strftime('%x %X %Z')}")
+                tooltip_lines.append(f"{m.value}: {format_health(m, health, True)}")
+            tooltip_lines.append(f"Time stamp: {health.timestamp.strftime('%x %X %Z')}")
```
[the remark this answers is not in the document]
Comment (mateuszmidor) [reply presentation/matrix_view.py:260]: Updating.

File: presentation/matrix_view.py [hunk 2]
```diff
@@ -273,54 +256,10 @@ def make_cell_hover_text(self, from_agent: Agent, to_agent: Agent, mesh: MeshRes
...
-        from_agent = self._agents.get_by_name(click_data["points"][0]["y"])
-        logging.debug(
-            "click: x: %s y: %s from: %s to: %s",
-            click_data["points"][0]["x"],
-            click_data["points"][0]["y"],
-            from_agent.id,
-            to_agent.id,
-        )
-        return from_agent.id, to_agent.id
+        return "\n".join(tooltip_lines)
```
[the remark this answers is not in the document]
Comment (mateuszmidor) [reply presentation/matrix_view.py:265]: Updating

File: presentation/matrix_view.py [hunk 3]
```diff
@@ -138,133 +158,96 @@ def make_matrix_content(self, results: MeshResults, config: MeshConfig, metric:
...
-            sla_levels.append(sla_levels_col)
-
-        return sla_levels
-
-    def get_thresholds(self, metric: MetricType) -> Thresholds:
+                        row.append(MatrixCell(text="-", tooltip=tooltip, tag=CellTag.NODATA, href=href))
+            rows.append(row)
+        return rows
+
+    def _tag_to_color(self, tag: CellTag) -> MatrixCellColor:
```
[the remark this answers is not in the document]
Comment (mateuszmidor) [reply presentation/matrix_view.py:226]: Updating

File: presentation/matrix_view.py [hunk 4]
```diff
@@ -1,55 +1,78 @@
...
+    for line in text.split("\n"):
+        result.append(line)
+        result.append(html.Br())
+    return result
+
+
+def cell_tag(val: MetricValue, warning_threshold: Threshold, critical_threshold: Threshold) -> CellTag:
+    if math.isnan(val):
+        return CellTag.NODATA
+    if val < warning_threshold:
```
[the remark this answers is not in the document]
Comment (mateuszmidor) [reply presentation/matrix_view.py:52]: Updating

File: presentation/matrix_view.py [hunk 5]
```diff
@@ -1,55 +1,78 @@
...
+@dataclass
+class MatrixCell:
+    text: str = ""
+    href: str = ""
+    tooltip: str = ""
+    tag: CellTag = CellTag.NODATA
 
 
 def agent_label(agent: Agent) -> str:
-    return agent.name
```
[the remark this answers is not in the document]
Comment (mateuszmidor) [reply presentation/matrix_view.py:42]: Adding configurable agent label formatting string

**Review by mmac-m3a — APPROVED** [review]
We still need to fix agent names (we are not getting the right ones at the moment), but it can be done separately.

**Pull request merged**